components/org.wso2.identity.webhook.common.event.handler/src/main/java/org/wso2/identity/webhook/common/event/handler/LoginEventHookHandler.java

```diff
@@ -0,0 +1,189 @@
+                IdentityEventConstants.EventName.AUTHENTICATION_FAILURE.name().equals(eventName);
+    }
+
+    @Override
+    public void handleEvent(Event event) throws IdentityEventException {
+
+        EventData eventData = buildEventDataProvider(event);
+
+        if (eventData.getAuthenticationContext().isPassiveAuthenticate()) {
+            return;
+        }
+
+        //TODO: Add the implementation to read the Event Schema Type from the Tenant Configuration
+        LoginEventPayloadBuilder payloadBuilder = PayloadBuilderFactory
+                .getLoginEventPayloadBuilder(EVENT_SCHEMA_TYPE_WSO2);
+        EventAttribute loginEventPublisherConfig = getLoginEventPublisherConfigForTenant(
+                eventData.getAuthenticationContext().getLoginTenantDomain(), event.getEventName());
+        EventPayload eventPayload;
+        String eventUri;
+
+        if (IdentityEventConstants.EventName.AUTHENTICATION_SUCCESS.name().equals(event.getEventName()) &&
+                loginEventPublisherConfig.isPublishEnabled()) {
+            eventPayload = payloadBuilder.buildAuthenticationSuccessEvent(eventData);
+            eventUri = EventHookHandlerUtils.getEventUri(Constants.EventHandlerKey.LOGIN_SUCCESS_EVENT);
+            String tenantDomain = eventData.getAuthenticationContext().getLoginTenantDomain();
+            SecurityEventTokenPayload securityEventTokenPayload = buildSecurityEventToken(eventPayload,
+                    eventData.getAuthenticationContext(), eventUri);
+            EventHookHandlerUtils.publishEventPayload(securityEventTokenPayload, tenantDomain, eventUri);
+        } else if (IdentityEventConstants.EventName.AUTHENTICATION_STEP_FAILURE.name().equals(event.getEventName()) &&
+                loginEventPublisherConfig.isPublishEnabled()) {
+            eventPayload = payloadBuilder.buildAuthenticationFailedEvent(eventData);
+            eventUri = EventHookHandlerUtils.getEventUri(Constants.EventHandlerKey.LOGIN_FAILED_EVENT);
+            String tenantDomain = eventData.getAuthenticationContext().getLoginTenantDomain();
+            SecurityEventTokenPayload securityEventTokenPayload = buildSecurityEventToken(eventPayload,
+                    eventData.getAuthenticationContext(), eventUri);
+            EventHookHandlerUtils.publishEventPayload(securityEventTokenPayload, tenantDomain, eventUri);
+        }
+    }
+
+    /**
+     * Check whether the login event handler is enabled.
+     *
+     * @return True if the login event handler is enabled.
+     */
+    public boolean isLoginEventHandlerEnabled() {
+
+        String enablePropertyKey = Constants.LOGIN_EVENT_HOOK_NAME + "." + Constants.ENABLE;
+        return this.configs != null && this.configs.getModuleProperties() != null &&
+                Boolean.parseBoolean(configs.getModuleProperties().getProperty(enablePropertyKey));
+    }
+
+    private EventAttribute getLoginEventPublisherConfigForTenant(String tenantDomain, String eventName) {
```

Comment:
Lets add method comments for all methods other than getters and setter. For getters and setters, it generates comment when generate via IDE.

Reply:
Well I think we don't need for all. But make sure we have method comments for public methods

Reply:
As per the conversation we had on https://github.com/wso2/carbon-identity-framework/pull/5810/files/22aed58101e07cf32f0afaf47ff475428b19d4e4#r1693035948 let's skip method comments